pilot/pkg/xds/ads.go

```diff
@@ -321,6 +315,10 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
 	// reqChannel and the connection not being enqueued for pushes to pushChannel until the
 	// initialization is complete.
 	<-con.initialized
+	// authorize client
```

Comment:
Should n't this be called before we start go routine for receive? May be move authenticate and authorize to initConnection?

Reply:
no, the receive go routine initialize proxy, authorize depend on it.

Reply:
Ah. Got it